app/src/androidTest/java/ch/epfl/polybazaar/FillListingActivityTest.java

```diff
@@ -240,6 +248,63 @@ public void cancelTakingPicture() throws Throwable {
     }
 
 
+    @Test
+    public void DialogAppearsWhenNoConnection() throws Throwable {
+        Intents.init();
+        useMockNetworkState(false);
+        useMockDataStore();
+        fillListing();
+        runOnUiThread(() -> fillSaleActivityTestRule.getActivity().findViewById(R.id.submitListing).performClick());
+        Thread.sleep(5000);
```

Comment:
You probably do not need all these waits with the Mock database, and even if you do they can be much much shorter. (I know I left one Thread.sleep in one of my test from 2 weeks ago as there was no mock database yet, so if you could delete it as well it would be nice).
The same comment applies to all the Thread.sleep